src/utils/geometry.rs

```diff
@@ -725,6 +725,29 @@ impl<N: Coordinate, Kind> Rectangle<N, Kind> {
     }
 }
 
+impl<Kind> Rectangle<f64, Kind> {
+    /// Convert to i32 for integer-space manipulations by rounding float values
+    #[inline]
+    pub fn to_i32_round<N: Coordinate>(self) -> Rectangle<N, Kind> {
+        Rectangle {
+            loc: self.loc.to_i32_round(),
+            size: self.size.to_i32_round(),
+        }
+    }
+
+    /// Convert to i32 for integer-space manipulations by flooring float values
+    #[inline]
+    pub fn to_i32_floor<N: Coordinate>(self) -> Rectangle<N, Kind> {
+        Rectangle::from_extemities(self.loc.to_i32_floor(), (self.loc + self.size).to_i32_ceil())
+    }
+
+    /// Convert to i32 for integer-space manipulations by ceiling float values
+    #[inline]
+    pub fn to_i32_ceil<N: Coordinate>(self) -> Rectangle<N, Kind> {
+        Rectangle::from_extemities(self.loc.to_i32_ceil(), (self.loc + self.size).to_i32_floor())
+    }
```

Comment:
I think those method should have more explicit names, or at least more explicit documentation. Their current behavior is the opposite from my I would have intuited given their names: it seemed to me natural to expect that ceil-ing a rectangle would yield a bigger rectangle than floor-ing it.

But overall, I'd say we'd better avoid using the words floor/ceil. What about `to_i32_grow` and `to_i32_shrink` for example?

Reply:
Oh right, I totally swapped them around. But that just shows, that they should be better docs and probably better names.

Reply:
I hope its cleaner now?

Reply:
Yep, that looks good now :+1: